### PR TITLE
Item extensions

### PR DESCRIPTION
**Pull request by AlexTugarev**
Also-by: Dennis Nobel <d.nobel@external.telekom.de>
Signed-off-by: Alex Tugarev <telekom@tugarev.de>

### CODE REVIEW

File: bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ManagedItemProvider.java [hunk 1]
```diff
@@ -70,6 +57,30 @@ public PersistedItem(String itemType, List<String> groupNames, String baseItemTy
...
+        if (recursive && item instanceof GroupItem) {
+            List<String> members = getMemberNamesRecursive((GroupItem) item, getAll());
+            for (String member : members) {
+                this.remove(member);
+            }
+        }
+        this.remove(item.getName());
+    }
+    
+    private List<String> getMemberNamesRecursive(GroupItem groupItem, Collection<Item> allItems) {
```
Comment (AlexTugarev) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ManagedItemProvider.java]: rename to getMemberNamesRecursively or getAllMemberNames ?

File: bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ActiveItem.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
+package org.eclipse.smarthome.core.items;
```
Comment (AlexTugarev) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ActiveItem.java]: license header

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/setup/ThingSetupManager.java [hunk 3]
```diff
@@ -0,0 +1,474 @@
...
+ * The {@link ThingSetupManager} provides various method to manage things. In
+ * contrast to the {@link ThingRegistry}, the {@link ThingManager} also creates
+ * Items and Links automatically and removes it, when the according thing is
+ * removed.
+ */
+public class ThingSetupManager {
+
+    public static final String TAG_CHANNEL_GROUP = "channel_group";
+    public static final String TAG_HOME_GROUP = "home_group";
+    public static final String TAG_THING = "thing";
```
Comment (AlexTugarev) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/setup/ThingSetupManager.java]: Shouldn't these tags be better defined in some kind of common tag collection?

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/AbstractLinkRegistry.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+ * 
+ * @author Dennis Nobel - Initial contribution
+ *
+ * @param <L>
+ *            Concrete type of the abstract link
+ */
+public abstract class AbstractLinkRegistry<L extends AbstractLink> extends AbstractRegistry<L, String> {
+
+    /**
+     * Returns if an item for a given item is linked to a channel or thing for a
```
Comment (AlexTugarev) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/AbstractLinkRegistry.java]: Typo.

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/AbstractLinkRegistry.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+            if (link.getUID().equals(uid) && link.getItemName().equals(itemName)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Returns the item name, which is bound to the given UID.
```
Comment (AlexTugarev) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/AbstractLinkRegistry.java]: Typo

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/AbstractLinkRegistry.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+            if (link.getUID().equals(uid)) {
+                return link.getItemName();
+            }
+        }
+        return null;
+    }
+    
+
+    /**
+     * Returns the item name, which is bound to the given UID.
```
Comment (AlexTugarev) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/AbstractLinkRegistry.java]: Typo.

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/setup/ThingSetupManager.java (hunk 3 above)
Comment (dnobel) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/setup/ThingSetupManager.java]: Yes, but I would like to introduce that later.

**Pull request closed without merging**